include/megaapi.h

```diff
@@ -2125,13 +2132,93 @@ class MegaStringList
      * @param i Position of the string that we want to get for the list
      * @return string at the position i in the list
      */
-    virtual const char* get(int i);
+    virtual const char* get(int i) const;
 
     /**
      * @brief Returns the number of strings in the list
      * @return Number of strings in the list
      */
-    virtual int size();
+    virtual int size() const;
+};
+
+class MegaStringListMap
+{
+public:
+    virtual ~MegaStringListMap();
+
+    static MegaStringListMap* createInstance();
+
+    virtual MegaStringListMap* copy() const;
+
+    /**
+     * @brief Returns the string list at the given key in the map
+     *
+     * The MegaStringMap retains the ownership of the returned string list. It will be only valid until
+     * the MegaStringMap is deleted.
+     *
+     * If the key is not found in the map, this function returns NULL.
+     *
+     * @param key Key to lookup in the map. Must be null-terminated
+     * @return String list at the given key in the map
+     */
+    virtual const MegaStringList* get(const char* key) const;
+
+    /**
+     * @brief Sets a value in the map for the given key.
+     *
+     * If the key already exists, the value will be overwritten by the
+     * new value.
+     *
+     * The map does not take ownership of the parameters passed, it makes
+     * a local copy.
+     *
+     * @param key The key in the map. It must be a null-terminated string.
+     * @param value The new value for the key in the map.
+     */
+    virtual void set(const char* key, const MegaStringList* value);
+
+    /**
+     * @brief Returns the number of (string, string list) pairs in the map
+     * @return Number of pairs in the map
+     */
+    virtual int size() const;
+};
+
+class MegaStringTable
+{
+public:
+    virtual ~MegaStringTable();
+
+    static MegaStringTable *createInstance();
+
+    virtual MegaStringTable* copy() const;
+
+    /**
+     * @brief Appends a new string list to the end of the table
+     *
+     * The parameter passed is copied.
```

Comment:
I think most clients would construct the string list specifically for this call, and then would have to delete it after passing it to this function, meaning we are forcing a second copy to be created.   Wouldn't it be more efficient to take ownership of the parameter, and if the client wants to keep and use the original, they can easily just call copy() on the list.

Reply:
But then shouldn't we be consistent with other data structures in this module as well. Shouldn't they take ownerships of all elements? Like StringList should then also take ownership etc

Reply:
After discussion IRL, will make the two new data structures take ownership of the string list passed.